lib/thrift/parser/models.ex

```diff
@@ -254,6 +254,31 @@ defmodule Thrift.Parser.Models do
 
       %Union{name: name, fields: fields}
     end
+
+    def validator(%Union{}, var_name) do
+      union_var = Macro.var(var_name, nil)
+      quote do
+        set_fields = unquote(union_var)
+        |> Map.delete(:__struct__)
+        |> Enum.reject(fn {_, val} -> is_nil(val) end)
+        case set_fields do
+          [] ->
+            :ok
+
+          [_] ->
+            :ok
+
+          set_fields ->
+            raise Thrift.Union.TooManyFieldsSetException.new(set_fields)
+        end
+      end
+    end
```

Comment:
I think that's a neat idea.

Building on it, I wonder if there's a maximum union "size" (field count) where it becomes less efficient to pattern match and more efficient to handle within a single function body. Because we control code generation, we could select the more optimal strategy.